Assets/MirageSDK/Plugins/WalletConnectSharp.Unity/Network/Client/Implementation/WebGLWebSocket.cs

```diff
@@ -92,16 +92,24 @@ public int GetInstanceId()
 			return _instanceId;
 		}
 
-		public UniTask Connect()
+		public async UniTask Connect()
 		{
+			UnityEngine.Debug.Log($"Calling WebSocket connect for {_instanceId}");
 			var ret = WebSocketConnect(_instanceId);
+			UnityEngine.Debug.Log($"WebSocketConnect returned {ret}");
 
 			if (ret < 0)
 			{
 				throw WebSocketHelpers.GetErrorMessageFromCode(ret, null);
 			}
 
-			return UniTask.CompletedTask;
+			while(State != WebSocketState.Open)
+			{
+				await UniTask.Delay(50);
```

Comment:
do we want to hardcode 50? 
Extract into readonly var: TimeSpan.FromSeconds(50) and use delay from timespan